internal/persistence/sql/relationtuples.go

```diff
@@ -200,6 +200,33 @@ func (p *Persister) DeleteRelationTuples(ctx context.Context, rs ...*relationtup
 	})
 }
 
+func (p *Persister) DeleteAllRelationTuples(ctx context.Context, query *relationtuple.RelationQuery) error {
+	return p.Transaction(ctx, func(ctx context.Context, c *pop.Connection) error {
+		sqlQuery := p.QueryWithNetwork(ctx)
+		if query.Namespace != "" {
+			n, err := p.GetNamespaceByName(ctx, query.Namespace)
+			if err != nil {
+				return err
+			}
+			sqlQuery.Where("namespace_id = ?", n.ID)
+		}
+		if query.Object != "" {
+			sqlQuery.Where("object = ?", query.Object)
+		}
+		if query.Relation != "" {
+			sqlQuery.Where("relation = ?", query.Relation)
+		}
+		if s := query.Subject(); s != nil {
+			if err := p.whereSubject(ctx, sqlQuery, s); err != nil {
+				return err
+			}
+		}
```

Comment:
Now this looks way better, but it is the same code as in the `GetRelationTuples` function.
Can you instead add a helper function similar to this (naming is just a draft):

```go
// whereQuery takes a relation tuple query and appends where clauses to the base pop query
func whereQuery(ctx context.Context, baseQuery *pop.Query, rq *relationtuple.RelationQuery) (*pop.Query, error) {
	if query.Namespace != "" {
		n, err := p.GetNamespaceByName(ctx, rq.Namespace)
		if err != nil {
			return nil, err
		}
		baseQuery.Where("namespace_id = ?", n.ID)
	}
	if rq.Object != "" {
		baseQuery.Where("object = ?", rq.Object)
	}
	if rq.Relation != "" {
		baseQuery.Where("relation = ?", rq.Relation)
	}
	if s := rq.Subject(); s != nil {
		if err := p.whereSubject(ctx, baseQuery, s); err != nil {
			return err
		}
	}
}
```

Reply:
And then of course use it in both functions.

Reply:
Yeah, I will update it.